src/openms/include/OpenMS/QC/TIC.h

```diff
@@ -71,7 +71,16 @@ namespace OpenMS
     @param bin_size RT bin size in seconds
     @return TIC Chromatogram
     **/
-    void compute(const MSExperiment &exp, float bin_size=0);
+    struct Result
+    {
+      std::vector<UInt> intensities;
+      std::vector<float> retention_times;
+      UInt area = 0;
+      UInt fall = 0;
+      UInt jump = 0;
+    };
+
+    Result compute(const MSExperiment& exp, float bin_size=0);
```

Comment:
since result is returned here,
`std::vector<MSChromatogram> results_;` is not needed anymore?

Reply:
mhh... wait, its probably needed?
now its a bit confusing what the other `results_` is... maybe document that a little...

Reply:
the Result struct does not contain the MSChromatogram returned by getTIC. Instead RTs and intensities are stores separately in vectors. Add std::vector<MSChromatogram> to the Result struct and get rid of results_? What happens with TIC::getResults()?

Reply:
removed the results_ vector and getResults function. The QualityControl tool now collects TIC::Result structures in a vector, which is used by TIC::addMetaDataMetricsToMzTab.
Is there a case where multiple TICs are added to MzTab at once?